apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRGenTask.groovy

```diff
@@ -40,7 +42,8 @@ public class ApolloIRGenTask extends NodeTask {
     dependsOn(ApolloCodeGenInstallTask.NAME)
 
     possibleGraphQLPaths = buildPossibleGraphQLPaths()
-    outputDir = project.file("src/$variant/graphql/generatedIR")
+    schemaFile = userProvidedSchemaFile() ?: searchForSchemaFile()
+    outputDir = schemaFile.getParentFile()
```

Comment:
A consequence of this is that IR Files for all variants will be under `src/main/graphql/...wherever_the_schema_file_was`.

A better alternative imo would be doing the same check we do in `GraphQlCompiler` for a structure matching `src/x/graphql/**` and generating the schema file in the directory that results after replacing x with the variant name.